active/0000-plaforms.md

```diff
@@ -0,0 +1,328 @@
+
+You can import the corresponding `app` version of a module into the platform addon.
+
+```js
+import Foo from '<project-name>/routes/foo';
+
+// our platform version of routes/foo
+export default Foo.extend({
+  hello: 'World!'
+});
+```
+
+### Platform Specific Config
+
+A platform config file is generated automatically for each platform you create with `ember platform`.
+```
+<project>
+  /config
+    /platforms
+      /<type>[-<name>].js
+```
+
+This configuration will be merged with your primary configuration.
+
+## Plugins
+
+Users should be able to create platform plugins as addons which implement 
+any number of specified hooks. These plugins should carry in the same tradition
+in form, function and hook names as `ember-cli-deploy`.
+
+## Creating Plugins
+
+Platforms have several forms of plugins:
+
+1. `process hooks` which add behaviors to one or more platforms.
+2. `platform plugins` which implement support and generators for a specific platform.
+3. `deploy plugins` which implement support for deployment of a platform.
+
+Platform plugins (generally) are also process hooks.
+
+## All plugins
+
+Platform plugins are ember-cli addons with a few platform specific traits:
+
+1. they contain a package.json keyword to identify them as plugins
+2. they are named ember-cli-platform-*
+3. they return an object that implements one or more of the ember-cli-platforms pipeline hooks
+
+
+### Creating a Process Hook
+
+#### Create an addon
+
+```cli
+ember addon ember-cli-platform-<name>
+```
+
+#### Mark the addon as a plugin
+
+Identify your addon as a plugin by updating the package.json like so:
+
+```js
+// package.json
+
+"keywords": [
+  "ember-addon",
+
+  // if present, indicates this addon contains process hooks for ember-cli-platforms
+  "ember-cli-platform-plugin",
+  
+  // if present, indicates this addon contains a platform implementation for ember-cli-platforms
+  "ember-cli-platform",
+  
+  // if present, indicates this addon implements deployment hooks
+  "ember-cli-platform-deploy"
+]
+```
+
+
+#### Implement one or more pipeline hooks
+
+In order for a plugin to be useful it must implement one or more of the ember-cli-platforms
+pipeline hooks.
+
+To do this you must implement a function called `createPlatformPlugin` in your `index.js` file.
+This function must return an object that contains:
+
+1. A name property which is what your plugin will be referred to.
+2. One or more implemented pipeline hooks
+
+**Example:**
+
+```js
+module.exports = {
+  name: 'ember-cli-platform-live-reload',
+
+  createPlatformPlugin: function(options) {
+    return {
+      name: "live-reload",
+
+      didBuild: function(context) {
+        //do something amazing here once the project has been built
+      }
+    };
+  }
+
+};
+```
+
+#### Process Hooks
+
+- configure
+- setup
+- willBuild
+- build
+- didBuild
+- teardown
+
+
+
+### Creating a Deployment Plugin
```

Comment:
We should limit the scope and leave this to an addon for the moment.

While this is mostly relevant for `ember-cli-deploy`, it's important to note a lot of base code is going to get repurposed for this project from ember-cli-deploy.